packages/eslint-config-typescript/requiring-type-checking.js

```diff
@@ -53,13 +53,23 @@ module.exports = {
         */
 
         /*
-            The default configuration for this rule requires that enum members
-            follow the camelCase style, which doesn't match the existing convention
-            within NI and the broader TS community to use PascalCase. The rest of the
-            configuration is identitical to the default.
+            Most of this configuration is identitical to the default with exceptions called out in comments below
         */
         '@typescript-eslint/naming-convention': [
             'error',
+            // Allow properties with custom-element-names, commonly used in HTMLElementNameTagMap
+            {
+                selector: 'property',
+                format: null,
+                filter: {
+                    // custom element regex from https://html-validate.org/rules/element-name.html
+                    regex: '[a-z][a-z0-9\\-._]*-[a-z0-9\\-._]*$',
```

Comment:
Note, this requires a dash (-). Technically, I don't think that is required.

Reply:
A dash is required for `custom elements` as the browser feature (names without `-` are reserved for HTML). Are there use cases to increase the scope outside of custom elements (don't think so, just checking if there was something on your mind)?

Reply:
No, I wasn't aware of that requirement. Thanks.